src/test/regress/sql/failure_single_mod.sql

```diff
@@ -0,0 +1,60 @@
+SELECT citus.mitmproxy('conn.allow()');
+SELECT citus.clear_network_traffic();
+
+SET citus.shard_count = 2;
+SET citus.shard_replication_factor = 2;
+
+CREATE TABLE mod_test (key int, value text);
+SELECT create_distributed_table('mod_test', 'key');
+
+-- verify behavior of single INSERT; should mark shard as failed
+SELECT citus.mitmproxy('conn.onQuery(query="^INSERT").kill()');
+INSERT INTO mod_test VALUES (2, 6);
+
+SELECT COUNT(*) FROM mod_test WHERE key=2;
+
+-- some clean up
+UPDATE pg_dist_shard_placement SET shardstate = 1
+WHERE shardid IN (
+  SELECT shardid FROM pg_dist_shard WHERE logicalrelid = 'mod_test'::regclass
+) AND shardstate = 3 RETURNING placementid;
+TRUNCATE mod_test;
+
+-- verify behavior of UPDATE ... RETURNING; should mark as failed
+INSERT INTO mod_test VALUES (2, 6);
```

Comment:
This insert fails already and marks the shard on the mitm-ed machine as failed. causing this test to not verify the update behaviour

```suggestion
SELECT citus.mitmproxy('conn.allow()');
INSERT INTO mod_test VALUES (2, 6);
```